Code/GraphMol/Canon.cpp

```diff
@@ -20,12 +19,11 @@ namespace RDKit {
 namespace Canon {
 namespace {
 bool isUnsaturated(const Atom *atom, const ROMol &mol) {
-  ROMol::OEDGE_ITER beg, end;
-  boost::tie(beg, end) = mol.getAtomBonds(atom);
-  while (beg != end) {
-    const Bond *bond = mol[*beg];
-    ++beg;
-    if (bond->getBondType() != Bond::SINGLE) {
+  for (const auto &bndItr :
+       boost::make_iterator_range(mol.getAtomBonds(atom))) {
+    // can't just check for single bonds, because dative bonds also have an
+    // order of 1
+    if (mol[bndItr]->getBondTypeAsDouble() > 1.001) {
```

Comment:
1.0 is perfectly representable in floating point.  I think you can just do getBondTypeAsDouble() > 1

Reply:
agreed